src/cdk/tree/control/nested-tree-control.ts

```diff
@@ -41,10 +41,15 @@ export class NestedTreeControl<T> extends BaseTreeControl<T> {
   /** A helper function to get descendants recursively. */
   protected _getDescendants(descendants: T[], dataNode: T): void {
     descendants.push(dataNode);
-    this.getChildren(dataNode).pipe(take(1)).subscribe(children => {
-      if (children && children.length > 0) {
-        children.forEach((child: T) => this._getDescendants(descendants, child));
-      }
-    });
+    const childrenNodes = this.getChildren(dataNode);
+    if (Array.isArray(childrenNodes)) {
+      childrenNodes.forEach((child: T) => this._getDescendants(descendants, child));
+    } else if (childrenNodes instanceof Observable) {
+      childrenNodes.pipe(take(1)).subscribe(children => {
```

Comment:
Since this is being done recursively and `childrenNodes` can be async, couldn't this potentially lead to some race conditions?